lib/shard/src/proxy_segment/mod.rs

```diff
@@ -320,6 +320,13 @@ impl ProxySegment {
 
         Ok(())
     }
+
+    pub(super) fn bump_version(&mut self, op: SeqNumberType) {
+        match &self.wrapped_segment {
+            LockedSegment::Original(rw_lock) => rw_lock.write().bump_segment_version(op),
+            LockedSegment::Proxy(rw_lock) => rw_lock.write().bump_version(op),
+        }
+    }
```

Comment:
In a proxy segment, the wrapped segment should remain unchanged. I'd argue the same is true for the segment version.

Let's bump the the write segment instead.

Reply:
Fixed!